magento/magento2
# Improved calculating version hash for the js-translation.json file.

### Description

Currently the version hash which is used for the js-translation.js file is derived from its modification time on the filesystem + its relative directory name.
We've noticed on a project which is being deployed over multiple webservers behind a loadbalancer, that the outputted version differs on all the nodes. This is because the modification timestamp of the file is off by a couple of seconds on all the nodes.
I haven't seen any negative effects of the different version hashes, but I think it would be nice if the outputted html from all nodes is 100% identical, which it isn't right now.

I've changed the logic, so instead of using the timestamp of the file, it uses the contents of the file to calculate the version hash. This way, it will be the same on all the nodes and the outputted version will be the same every time you refresh your browser (and hit other nodes).

This problem occurs on both Magento CE 2.1.7 and 2.2.0-RC1.6

This pull request is a first quick version, to see if there is interest in this. I think the logic of calculating the hash should be moved over to the Block itself, instead of being done in the phtml file.
There should probably be a few tests added as well.
I'm not sure if this breaks the backwards compatibility rules, since it adds a public method to an `@api` annotated class?
If there is interest in this change, I'll try to improve this PR further.

### Fixed Issues (if relevant)
None that I could find

### Manual testing scenarios
1. Setup a new Magento CE 2.1.7 or 2.2.0-RC1.6 installation
2. View the html source of the homepage, you'll see a line:
`if (versionObj.version !== '6c7483a300fdf2968a0d04aea293e882d9f5bf1e') {`
3. Run `touch pub/static/frontend/Magento/luma/en_US/js-translation.json` so the modification timestamp of the file is changed
4. Flush your caches
5. View the html source of the homepage again, the version hash is changed, even though the file in itself wasn't changed

### Contribution checklist
